src/adaptors/in-memory-articles.ts

```diff
@@ -0,0 +1,33 @@
+import { JsonLdObj, Iri } from 'jsonld/jsonld-spec';
+import { Articles } from './articles';
+
+export default class InMemoryArticles implements Articles {
+    private nodes: { [key: string]: JsonLdObj } = {};
+
+    * [Symbol.iterator](): Iterator<JsonLdObj> {
+      yield* Object.values<JsonLdObj>(this.nodes);
+    }
+
+    async count(): Promise<number> {
+      return Object.values(this.nodes).length;
+    }
+
+    async get(id: Iri): Promise<JsonLdObj> {
+      if (this.has(id)) {
+        return this.nodes[id];
+      }
+      throw new RangeError(`An item with the ID ${id} could not be found`);
```

Comment:
I'd prefer reversing this, generally throwing an error shouldn't be the default path (better to fail quickly).